x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/TransportGetSearchApplicationAction.java

```diff
@@ -52,6 +54,16 @@ protected void doExecute(
                     for (String key : inconsistentIndices.keySet()) {
                         HeaderWarning.addWarning(key + " " + inconsistentIndices.get(key));
                     }
+                    if (searchApplication.hasStoredTemplate() == false) {
+                        HeaderWarning.addWarning(SearchApplication.NO_TEMPLATE_STORED_WARNING);
+                        return new GetSearchApplicationAction.Response(
```

Comment:
Would returning the default template in the get() method avoid this case distinction? Or maybe we can do that on the toXContent() method for the search application?

Reply:
No, because we want to override the stored `null` with the default template

Reply:
Using the getter in toXContent would get this done?

What I'm trying is to get rid of that `if`. If we use the getter and the getter returns the stored template or the default template in case it's not defined, could we get rid of this and make it simpler?

Reply:
Yeah, I can get rid of the if.

Reply:
Done. [commit](https://github.com/elastic/elasticsearch/pull/98138/commits/2471ec27ff5900185b41a192b74ebf5c48a1bbdc)